src/vc/VerifiableCredential.ts

```diff
@@ -1,10 +1,128 @@
+        throw new Error('Evidence id is not a valid URL');
+    }
+    if (!_.isArray(evidenceItem.type)) {
+        throw new Error('Evidence type is not an Array object');
+    }
+}
+
+function transformMetaConstraint(constraintsMeta: any) {
+    const resultConstraints: any[] = [];
+
+    // handle special field constraints.meta.credential
+    const constraintsMetaKeys = _.keys(constraintsMeta.meta);
+    _.forEach(constraintsMetaKeys, (constraintKey: any) => {
+        const constraint = constraintsMeta.meta[constraintKey];
+        const siftConstraint = {};
+        // handle special field constraints.meta.credential
+        if (constraintKey === 'credential') {
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            siftConstraint.identifier = constraint;
+        } else if (constraint.is) {
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            siftConstraint[constraintKey] = constraint.is;
+        } else {
+            throw new Error(`Malformed meta constraint "${constraintKey}": missing the IS`);
+        }
+        resultConstraints.push(siftConstraint);
+    });
+    return resultConstraints;
+}
+
+function serializeEvidence(evidence: any) {
+    const evidenceList = _.isArray(evidence) ? evidence : [evidence];
+    return _.map(evidenceList, (evidenceItem: any) => {
+        validateEvidence(evidenceItem);
+        return {
+            id: evidenceItem.id,
+            type: evidenceItem.type,
+            verifier: evidenceItem.verifier,
+            evidenceDocument: evidenceItem.evidenceDocument,
+            subjectPresence: evidenceItem.subjectPresence,
+            documentPresence: evidenceItem.documentPresence,
+        };
+    });
+}
+
+function transformConstraint(constraints: any) {
+    const resultConstraints: any[] = [];
+
+    _.forEach(constraints.claims, (constraint: any) => {
+        if (!constraint.path) {
+            throw new Error('Malformed contraint: missing PATTH');
+        }
+        if (!constraint.is) {
+            throw new Error('Malformed contraint: missing IS');
+        }
+
+        const siftConstraint = {};
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        siftConstraint[constraint.path] = constraint.is;
+        resultConstraints.push(siftConstraint);
+    });
+
+    return resultConstraints;
+}
+
+function transformDate(obj: any) {
+    return new Date(obj.year, (obj.month - 1), obj.day).getTime() / 1000;
+}
+
+function isDateStructure(obj: any) {
+    const objKeys = _.keys(obj);
+    if (objKeys.length !== 3) {
+        // it has more or less keys the (day, month, year)
+        return false;
+    }
+    return (_.includes(objKeys, 'day') && _.includes(objKeys, 'month') && _.includes(objKeys, 'year'));
+}
+
+const convertDeltaToTimestamp = (delta: any) => time.applyDeltaToDate(delta).getTime() / 1000;
```

Comment:
I think delta could be typed to `string`? since in `convertTimestampIfString` calls `convertDeltaToTimestamp` when it is a string